papers/ring_vrf/pedersen_vrf.tex

```diff
@@ -22,26 +22,28 @@ \section{Ring VRFs from the Pedersen VRF}
 %  \PedVRF consists of algorithms having superficially similar signatures.
 % \footnote{We do not define security for \PedVRF because pseudo-randomness becomes too interesting}
 
-We shall need $\ecE$ to be pairing friendly in later sections.
-Although not strictly essential, we prefer if \PreOut cannot be verified
-by pairings, mostly as a miss-use resistance measure.  We therefore
-also require an Edwards curve $\ecE'$ with a subgroup $\grE'$
-of the same order $p$ as $\grE_1$ and $\grE_2$.
+%NOTE: The below paragraph should not be here since we give here a general construction of a ring VRF which works securely in ANY group where DDH is hard. We can give these details in the end of the paper or somewhere else if it is necessary.
+%We shall need $\ecE$ to be pairing friendly in later sections.
+%Although not strictly essential, we prefer if \PreOut cannot be verified
+%by pairings, mostly as a miss-use resistance measure.  We therefore
+%also require an Edwards curve $\ecE'$ with a subgroup $\grE'$
+%of the same order $p$ as $\grE_1$ and $\grE_2$.
 
-We select a generator $\genG$ of $\grE$ as our public key base point by
-any desired method, but then fix a second generator $\genB$ of $\grE$ independent from $\genG$.
+Our construction works in a prime $ p $-order group $ \grE $. We use a generator $\genG$ of $\grE$ as our public key base point by
+any desired method, but then fix a second generator $\genB$ of $\grE$ independent from $\genG$. We deploy three random oracles $ H_{\grE}:\{0,1\}^* \rightarrow \grE $, $ H', H_p: \{0,1\}^* \rightarrow \F_p $. 
```

Comment:
We introduce the curves and hash functions in the background section, where Oana suggested [removing random oracle from the definition.](https://github.com/w3f/ring-vrf/commit/d42acea9b5a93b143ca55e8bf35057a07381fb99#diff-5dcb4cb4c588ef9a405c34101027e6bf7c0280edd8b2f753af8632d15b44372e)   I've previously noticed Raid Wahby criticize using random oracle this way too, saying "[H_{\grE}] is called a hash function to an elliptic curve" aka hash-to-curve. 

It's kinda fake implementation generality to talk about general groups here, as interesting ones like RSA groups need slightly different protocols.  There is some theory generality as you can likely prove security in GGM and weaker hash assumptions than RO.  Ain't too worried about that myself, but I can try to find a compromise, and we can discuss this further.

You suggest removing the cofactor above, but real implementation generality comes form explicitly handling the cofactor h like I do above.  We cannot afaik build prime order groups with hard DDH except using elliptic curves.  We can have deserialization code that enforced being inside some prime order subgroup, and this restriction is appropriate when doing pairings, but winds up much less general overall.

Reply:
A construction is a prime order group is less general than a construction in a composite order group.  We've ample history where omitting this detail causes real damage. 

If possible, this section should be about both how to implement it and a construction usable in security proofs.

Reply:
No, it should not